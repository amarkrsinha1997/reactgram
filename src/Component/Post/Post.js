import React from 'react'
import PropTypes from 'prop-types'
import { Image } from '../Base/Image/Image'
import { Tabs } from "../Base/Tabs/Tabs"
import './Post.scss';

const Post = props => {
  return (
    <div className="post">
      <div className="post-image">
        <Image imageName={props.display_src} alt={props.display_src} />
      </div>
      <Tabs>
        <Tabs.Button>
          Like
        </Tabs.Button>
        <Tabs.Button>
          Comments
        </Tabs.Button>
      </Tabs>
    </div>
  )
}

Post.propTypes = {

}

export { Post }
