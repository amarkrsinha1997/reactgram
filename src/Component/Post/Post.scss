@import "src/App.scss";

.post-image {
	width: 500px;
	@include mobileToPhablet {
		width: 100%;
		height: 100%;
	}
}

.post {
  display: flex;
	align-items: center;
	justify-content: space-between;
	flex-direction: column;
	flex-wrap: nowrap;
  margin: 10px;
	background-color: #cecece;
  max-width: 500px;
  @include mobileToTablet {
		width: 100%;
		height: 100%;
	}
}