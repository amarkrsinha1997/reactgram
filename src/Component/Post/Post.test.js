import React from "react";
import { Post } from './Post';
import { render, fireEvent  } from "@testing-library/react";

describe('Post', () => {
	let props;
	beforeEach(() => {
		props = {
			"code": "_A2r0aQcfD",
			"caption": "Some serious hardware meet JavaScript hacks going down this week at hackeryou. Excited for demo day!",
			"likes": 57,
			"id": "1135147611821557699",
			"display_src": "12276809_750065668431999_184252508_n.jpg",
			"comment": [
				{
					"text": "Uhu!",
					"user": "lucascaixeta",
					"id": 1
				}
			]
		};
	})
	test('should render properly', () => {
		const { container } = render(<Post {...props}/>);
		expect(container).toMatchSnapshot();
	});
	test('should be able to see comment on the post when clicked on comment tab', () => {
    const { queryByTestId, queryByText } = render(<Post {...props} />);
    const commentElements = queryByTestId(/_A2r0aQcfD/)
    expect(commentElements).toBeNull()
    
    const commentTabElement = queryByText(/Comments/)
    fireEvent.click(commentTabElement)
    expect(queryByTestId(/_A2r0aQcfD/)).toBeInTheDocument()
	})
	test.todo('should be able to like the post')
	test.todo('should be able to comment on the post')
});