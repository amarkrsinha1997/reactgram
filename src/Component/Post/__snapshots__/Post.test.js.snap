// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post should render properly 1`] = `
<div>
  <div
    class="post"
  >
    <div
      class="post-image"
    >
      <img
        alt="12276809_750065668431999_184252508_n.jpg"
        class="image"
        data-testid="12276809_750065668431999_184252508_n.jpg"
        src="https://raw.githubusercontent.com/pankajladhar/Reactgram/master/app/images/12276809_750065668431999_184252508_n.jpg"
      />
    </div>
    <div
      class="tabs"
    >
      <div
        class="seperator"
      >
        <div
          class="tab"
          role="button"
        >
          Like
          <style>
            
.tab {
	background: #dedede;
	padding: 10px 20px;
	color: #333333;
	text-align: center;
}

          </style>
        </div>
      </div>
      <div
        class="seperator"
      >
        <div
          class="tab"
          role="button"
        >
          Comments
          <style>
            
.tab {
	background: #dedede;
	padding: 10px 20px;
	color: #333333;
	text-align: center;
}

          </style>
        </div>
      </div>
    </div>
  </div>
</div>
`;
